src/platform_impl/linux/common/keymap.rs

```diff
@@ -0,0 +1,741 @@
+        164 => KeyCode::MediaPlayPause,
+        165 => KeyCode::MediaTrackPrevious,
+        166 => KeyCode::MediaStop,
+        // 167 => KeyCode::RECORD,
+        // 168 => KeyCode::REWIND,
+        // 169 => KeyCode::PHONE,
+        // 170 => KeyCode::ISO,
+        // 171 => KeyCode::CONFIG,
+        // 172 => KeyCode::HOMEPAGE,
+        // 173 => KeyCode::REFRESH,
+        // 174 => KeyCode::EXIT,
+        // 175 => KeyCode::MOVE,
+        // 176 => KeyCode::EDIT,
+        // 177 => KeyCode::SCROLLUP,
+        // 178 => KeyCode::SCROLLDOWN,
+        // 179 => KeyCode::KPLEFTPAREN,
+        // 180 => KeyCode::KPRIGHTPAREN,
+        // 181 => KeyCode::NEW,
+        // 182 => KeyCode::REDO,
+        183 => KeyCode::F13,
+        184 => KeyCode::F14,
+        185 => KeyCode::F15,
+        186 => KeyCode::F16,
+        187 => KeyCode::F17,
+        188 => KeyCode::F18,
+        189 => KeyCode::F19,
+        190 => KeyCode::F20,
+        191 => KeyCode::F21,
+        192 => KeyCode::F22,
+        193 => KeyCode::F23,
+        194 => KeyCode::F24,
+        // 200 => KeyCode::PLAYCD,
+        // 201 => KeyCode::PAUSECD,
+        // 202 => KeyCode::PROG3,
+        // 203 => KeyCode::PROG4,
+        // 204 => KeyCode::DASHBOARD,
+        // 205 => KeyCode::SUSPEND,
+        // 206 => KeyCode::CLOSE,
+        // 207 => KeyCode::PLAY,
+        // 208 => KeyCode::FASTFORWARD,
+        // 209 => KeyCode::BASSBOOST,
+        // 210 => KeyCode::PRINT,
+        // 211 => KeyCode::HP,
+        // 212 => KeyCode::CAMERA,
+        // 213 => KeyCode::SOUND,
+        // 214 => KeyCode::QUESTION,
+        // 215 => KeyCode::EMAIL,
+        // 216 => KeyCode::CHAT,
+        // 217 => KeyCode::SEARCH,
+        // 218 => KeyCode::CONNECT,
+        // 219 => KeyCode::FINANCE,
+        // 220 => KeyCode::SPORT,
+        // 221 => KeyCode::SHOP,
+        // 222 => KeyCode::ALTERASE,
+        // 223 => KeyCode::CANCEL,
+        // 224 => KeyCode::BRIGHTNESSDOW,
+        // 225 => KeyCode::BRIGHTNESSU,
+        // 226 => KeyCode::MEDIA,
+        // 227 => KeyCode::SWITCHVIDEOMODE,
+        // 228 => KeyCode::KBDILLUMTOGGLE,
+        // 229 => KeyCode::KBDILLUMDOWN,
+        // 230 => KeyCode::KBDILLUMUP,
+        // 231 => KeyCode::SEND,
+        // 232 => KeyCode::REPLY,
+        // 233 => KeyCode::FORWARDMAIL,
+        // 234 => KeyCode::SAVE,
+        // 235 => KeyCode::DOCUMENTS,
+        // 236 => KeyCode::BATTERY,
+        // 237 => KeyCode::BLUETOOTH,
+        // 238 => KeyCode::WLAN,
+        // 239 => KeyCode::UWB,
+        240 => KeyCode::Unidentified(NativeKeyCode::Unidentified),
+        // 241 => KeyCode::VIDEO_NEXT,
+        // 242 => KeyCode::VIDEO_PREV,
+        // 243 => KeyCode::BRIGHTNESS_CYCLE,
+        // 244 => KeyCode::BRIGHTNESS_AUTO,
+        // 245 => KeyCode::DISPLAY_OFF,
+        // 246 => KeyCode::WWAN,
+        // 247 => KeyCode::RFKILL,
+        // 248 => KeyCode::KEY_MICMUTE,
+        _ => KeyCode::Unidentified(NativeKeyCode::XKB(rawkey)),
```

Comment:
I've tried bringing this up before, but I may have explained it poorly. When a `keysym` isn't mapped by Winit, it is potentially useful to get the raw `keysym` in `Key::Unidentified`. When a `keycode` isn't mapped by Winit, it is potentially useful to get the raw `keycode` in `KeyCode::Unidentified`. `NativeKeyCode` only considers one of these cases, specifically the latter case. I think there may be a need to add a `NativeKeySym` enum.

> Otherwise if you get a `NativeKeyCode`, you don't know if the u32 inside is a rawkey or a keysym.

If all backends use it the way it's used here, then you'd know it based on whether you got it from a `Key::Unidentified` or a `KeyCode::Unidentified`. I think the Windows and macOS backends don't do this, however, and that's a bit of an oversight on my end.

Reply:
I agree that it's useful to be able to know both the keysym and the keycode. But isn't the conversion from keycode to keysym trivial when the key is not a "Character"?

Furthermore I'm still feeling very uneasy about storing two different things in a single type. It is true that the programmer can just keep in mind where did the variable come from in order to know what does it contain. But this gets hard if you have a large codebase and you look at a function like

```rust
fn do_something_funky(native_keycode: NativeKeyCode) {
    // Is `native_keycode` a keycode or a keysym?
}
```

I think that the most important thing here is to be able to *differentiate* between the Unidentified keys, which is possible no matter if we use a `keysym` or a `keycode`. But again, winit could provide a function that converts between these (for values that have a trivial mapping from one to the other)

Reply:
> But isn't the conversion from keycode to keysym trivial when the key is not a "Character"?

We can't rely on that since
- Support on the web for such a conversion will likely be non-existent or at the very least not supported in every major browser
- For such a conversion to be correct, you have to know the active keyboard layout at the time the keyboard event was received

> I think that the most important thing here is to be able to differentiate between the Unidentified keys, which is possible no matter if we use a `keysym` or a `keycode`. 

To me, the purpose of storing `keysym` *and* the `keycode` in an `Unidentified` variant on their respective enums is to say, "Sorry, we couldn't map this key. Here's the raw value we couldn't map.". In the case of `Key`, that raw value is the `keysym`.

Reply:
> To me, the purpose of storing `keysym` _and_ the `keycode` in an `Unidentified` variant on their respective enums is to say, "Sorry, we couldn't map this key. Here's the raw value we couldn't map.".

It does seem to be useful, which is why I wrote that I is useful to have the respective value handed downstream. But upon more thought I realized that I cannot come up with a specific use case for it. Can you? Let's say that the `Unidentified` variant contains a keysym for the `logical_key` and a keycode for the `physical_key`. What can downstream do with the keycode in the `physical_key` that they couldn't do if they instead received a keysym there?

Reply:
> What can downstream do with the keycode in the physical_key that they couldn't do if they instead received a keysym there?

Because reporting both allows both games *and* GUI-like applications to accept user-provided keybinds (through some configuration menu) for keys which winit doesn't map.
Games would like to use the physical key, and GUI-like applications (including some games/parts of games) would like to use the logical key for the value that triggers an action.

Reply:
I think I know how we can resolve this disagreement: by using separate variants for the different kinds of values.
```rust
pub enum NativeKeyCode {
    Unidentified,
    Windows(u16),
    XkbCode(u32),
    XkbSym(u32),
    MacOS(u32),
}
```

I would be happy with this because this way there's no ambiguity about what kind of value is contained in the variant.

Reply:
I'll do that for now in order to get this PR merged, but I'm *not* much of a fan of this approach, so I'll probably bring it back up later.

Reply:
Sounds good!

Reply:
It has been done.